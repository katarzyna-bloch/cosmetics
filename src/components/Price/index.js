import styled from 'styled-components'
import PropTypes from 'prop-types'

const OldPrice = styled.div`
  text-decoration: line-through;
  font-weight: 600;
  padding-right: 10px;
`

const PriceAfterDiscount = styled.div`
  font-weight: 600;
  color:  ${props => (props.hasDiscount ? 'red' : 'none')};
`

const Price = ({oldPrice, hasDiscount, price}) => {
  return (
  <>
    {oldPrice && (
      <OldPrice>{oldPrice}</OldPrice>
    )}
    <PriceAfterDiscount hasDiscount={hasDiscount}>{price}</PriceAfterDiscount>
  </>
  )
}

Price.propTypes = {
  oldPrice: PropTypes.string,
  hasDiscount: PropTypes.bool,
  price: PropTypes.string,
}

export default Price
