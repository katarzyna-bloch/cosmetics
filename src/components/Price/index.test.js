import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Price from '.'

it('should render correctly', () => {
  const props = {
    price: '15 zł',
    oldPrice: '50 zł',
    hasDiscount: true,
  }
  const wrapped = shallow(<Price {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
