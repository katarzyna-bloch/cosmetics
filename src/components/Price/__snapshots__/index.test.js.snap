// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<styled.div>
  <styled.div>
    15 zł
  </styled.div>
</styled.div>
`;

exports[`should render correctly with has discount 1`] = `
<styled.div>
  <styled.div>
    50 zł
  </styled.div>
  <styled.div
    hasDiscount={true}
  >
    15 zł
  </styled.div>
</styled.div>
`;
